packages/expo-modules-core/common/cpp/fabric/ExpoViewProps.cpp

```diff
@@ -17,10 +17,11 @@ std::unordered_map<std::string, folly::dynamic> propsMapFromProps(const ExpoView
 
   // Iterate over values in the raw props object.
   // Note that it contains only updated props.
-  rawProps.iterateOverValues([&propsMap](react::RawPropsPropNameHash hash, const char *name, const react::RawValue &value) {
-    std::string propName(name);
-    propsMap[propName] = (folly::dynamic)value;
-  });
+  const auto& dynamicRawProps = static_cast<folly::dynamic>(rawProps);
+  for (const auto& propsPair : dynamicRawProps.items()) {
+    std::string propName = propsPair.first.getString();
+    propsMap[propName] = (folly::dynamic)propsPair.second;
+  }
```

Comment:
In 0.76 `iterateOverValues` was removed from `RawPropsParser`, more details in https://github.com/facebook/react-native/pull/45088.  We can instead just directly cast `rawProps` using folly dynamic type and iterate over each key-value

Reply:
Such a pity it went into this direction. It would be much more beneficial for us if we get props as JSI values, not folly dynamic 😞

Reply:
yes maybe we can just copy the code at
https://github.com/facebook/react-native/blob/893a4b30b1596fff1a85e66139ec697674673034/packages/react-native/ReactCommon/react/renderer/core/RawPropsParser.cpp#L185-L204

thanks Gabriel for sharing where the changes coming from. it helps pretty much

Reply:
We can't – `mode_`, `value_` and `runtime_` are all private 😢

Reply:
There might be a way to achieve this by providing our own raw props parser in our custom component descriptor that would parse everything to JSI. Although, this is only the new architecture and we would have to keep the existing logic ([passing shared object id instead of the object as is](https://github.com/expo/expo/blob/5d9067166e3c8a3c8578228b59084a3e1f5e8baa/packages/expo-video/src/VideoView.tsx#L68-L77)) on the old architecture, so I refrained from implementing it.

Reply:
what a pity 😞
also in the original react-native pr, it said using dynamic parser is faster. not sure how it could be, maybe jsi round trips get it slow 🤔